x-pack/plugins/canvas/server/routes/custom_elements/create.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RouteInitializerDeps } from '../';
+import {
+  CUSTOM_ELEMENT_TYPE,
+  API_ROUTE_CUSTOM_ELEMENT,
+} from '../../../../../legacy/plugins/canvas/common/lib/constants';
+import { getId } from '../../../../../legacy/plugins/canvas/public/lib/get_id';
+import { CustomElementSchema } from './custom_element_schema';
+import { CustomElementAttributes } from './custom_element_attributes';
+import { okResponse } from '../ok_response';
+import { catchErrorHandler } from '../catch_error_handler';
+import { CustomElement } from '../../../../../legacy/plugins/canvas/types';
+
+export function initializeCreateCustomElementRoute(deps: RouteInitializerDeps) {
+  const { router } = deps;
+  router.post(
+    {
+      path: `${API_ROUTE_CUSTOM_ELEMENT}`,
+      validate: {
+        body: CustomElementSchema,
+      },
+      options: {
+        body: {
+          maxBytes: 26214400, // 25MB payload limit
+          accepts: ['application/json'],
+        },
+      },
+    },
+    catchErrorHandler(async (context, request, response) => {
+      if (!request.body) {
+        return response.badRequest({ body: 'A custom element payload is required' });
+      }
+
+      const customElement = request.body as CustomElement;
```

Comment:
Shouldn't need to cast this.  Body will be properly typed by whatever you define it as in validation